qemu/tests/qemu_guest_agent.py

```diff
@@ -1918,8 +1944,126 @@ def gagent_check_fsfreeze(self, test, params, env):
+        5) Check the FS is unfrozen from both guest agent side and
+         guest os side.
+
+        :param test: kvm test object
+        :param params: Dictionary with the test parameters
+        :param env: Dictionary with test environmen.
+        """
+        self._fsfreeze()
+
+    @error_context.context_aware
+    def gagent_check_fsfreeze_list(self, test, params, env):
+        """
+        Test guest agent commands "guest-fsfreeze-freeze-list"
+
+        Test steps:
+        1) Check the FS is thawed.
+        2) Freeze the FS without mountpoint.
+        3) Check the FS is frozen from both guest agent side and
+         guest os side.
+        4) Thaw the FS.
+        5) Check the FS is unfrozen from both guest agent side and
+         guest os side.
+        6) Freeze the FS with one valid mountpoint.
+        7) repeate step4-5.
+        8) Freeze the FS with two valid mountpoints
+        9) repeate step4-5.
+        8) Freeze the FS with one valid mountpoint and
+         one invalid mountpoint.
+        9) Check the result.
+
+        :param test: kvm test object
+        :param params: Dictionary with the test parameters
+        :param env: Dictionary with test environmen.
+        """
+        session = self._get_session(params, self.vm)
+        self._open_session_list.append(session)
+        image_size_stg0 = params["image_size_stg0"]
+
+        error_context.context("Format the new data disk and mount it.",
+                              logging.info)
+        if params.get("os_type") == "linux":
+            disk_data = list(utils_disk.get_linux_disks(session).keys())
+            mnt_point_data = utils_disk.configure_empty_disk(
+                session, disk_data[0], image_size_stg0, "linux",
+                labeltype="msdos")[0]
+            mount_points = ["/", mnt_point_data]
+        else:
+            disk_index = utils_misc.wait_for(
+                lambda: utils_disk.get_windows_disks_index(session,
+                                                           image_size_stg0),
+                120)
+            if disk_index:
+                logging.info("Clear readonly for disk and online it in"
+                             " windows guest.")
+                if not utils_disk.update_windows_disk_attributes(session,
+                                                                 disk_index):
+                    test.error("Failed to update windows disk attributes.")
+                mnt_point_data = utils_disk.configure_empty_disk(
+                    session, disk_index[0], image_size_stg0, "windows",
+                    labeltype="msdos")[0]
+                mount_points = ["C:\\", "%s:\\" % mnt_point_data]
+            else:
+                test.error("Didn't find any disk_index except system disk.")
+
+        error_context.context("Freeze fs without parameter of mountpoints.",
+                              logging.info)
+        self._fsfreeze(fsfreeze_list=True, check_mountpoints=mount_points)
```

Comment:
Since you have `mountpoints =  check_mountpoints if check_mountpoints` in `self._fsfreeze()`, so this line actually do the same thing with below `self._fsfreeze(fsfreeze_list=True, mountpoints=mount_points)`

Reply:
self._fsfreeze(fsfreeze_list=True, check_mountpoints=mount_points)
means freeze file system without parameter,the command like {"execute":"guest-fsfreeze-freeze-list" }, but check mountpoint need to figure out.

self._fsfreeze(fsfreeze_list=True, mountpoints=mount_points)
means freeze file system with parameter, like: {"execute":"guest-fsfreeze-freeze-list","arguments":{"mountpoints":["C:\\","D:\\"]}}, check_mountpoint is the mountpoints in 'arguments', no need to another figure out.

Reply:
@vivianQizhu Sorry to reply late.